openpype/hosts/tvpaint/plugins/publish/collect_instance_frames.py

```diff
@@ -34,3 +34,111 @@ def process(self, instance):
+        if keep_frame_index:
+            frame_start = (instance.context.data["sceneMarkIn"] + instance.context.data["sceneStartFrame"])
+            frame_end = (instance.context.data["sceneMarkOut"] + instance.context.data["sceneStartFrame"])
+            instance.data["keepFrameIndex"] = keep_frame_index
+            instance.data["frameStart"] = frame_start
+            instance.data["frameEnd"] = frame_end
+
+        # If custom frames are given
+        if custom_instance_frames:
+            # Create a list of custom frame to render
+            custom_frames = self.range_custom_frames(
+                        custom_instance_frames,
+                        context.data["sceneMarkIn"],
+                        context.data["sceneMarkOut"],
+                        context.data["sceneStartFrame"]
+                        )
+            # Avoid exporting frame before the tracker frameStart
+            if not all(frame >= asset_doc["data"]["frameStart"] for frame in custom_frames):
+                self.log.warning("The custom frames to export start BEFORE the scene Tracking Start Frame")
+                self.log.info("An auto clean will be applied to start at {}".format(asset_doc["data"]["frameStart"]))
+                # Remove frames lower that the tracker frameStart
+                custom_frames = [frame for frame in custom_frames if (asset_doc["data"]["frameStart"] <= frame)]
+                # Replace by the true tracker frameStart
+                custom_frames.insert(0, asset_doc["data"]["frameStart"])
+
+            # Avoid exporting frame before the scene sceneStartFrame
+            if not all(frame >= instance.context.data["sceneStartFrame"] for frame in custom_frames):
+                self.log.warning("The custom frames to export start BEFORE the tvpp scene Start Frame")
+                self.log.info("An auto clean will be applied to start at {}".format(instance.context.data["sceneStartFrame"]))
+                # Remove frames lower that the scene sceneStartFrame
+                custom_frames = [frame for frame in custom_frames if instance.context.data["sceneStartFrame"] <= frame]
+                # Replace by the true scene sceneStartFrame
+                custom_frames.insert(0, instance.context.data["sceneStartFrame"])
+
+            instance.data["customFrames"] = custom_frames
+
+            # Update the instance data
+            instance.data["frameStart"] = min(custom_frames)
+            instance.data["frameEnd"] = max(custom_frames)
+
+            if custom_instance_frames:
+                self.log.info("Export Custom frames {}".format(custom_frames))
+
+        if custom_instance_frames or keep_frame_index:
+            self.log.info("Changed frames Start/End {}-{} on instance {} ".format(frame_start, frame_end, instance.data["subset"]))
+
+    def range_custom_frames(self, custom_frames , mark_in, mark_out, sceneStartFrame):
```

Comment:
It's not a range it's a list of frames to export/publish

Reply:
Certes